src/systems/implicit_system.C

```diff
@@ -1222,8 +1222,16 @@ LinearSolver<Number> * ImplicitSystem::get_linear_solver() const
 
 std::pair<unsigned int, Real> ImplicitSystem::get_linear_solve_parameters() const
 {
-  return std::make_pair(this->get_equation_systems().parameters.get<unsigned int>("linear solver maximum iterations"),
-                        this->get_equation_systems().parameters.get<Real>("linear solver tolerance"));
+  if (parameters.have_parameter<unsigned int>("linear solver maximum iterations") &&
+     parameters.have_parameter<Real>("linear solver tolerance"))
+    return std::make_pair(parameters.get<unsigned int>("linear solver maximum iterations"),
+                          parameters.get<Real>("linear solver tolerance"));
+  else if (!parameters.have_parameter<unsigned int>("linear solver maximum iterations") &&
+     !parameters.have_parameter<Real>("linear solver tolerance"))
+    return std::make_pair(this->get_equation_systems().parameters.get<unsigned int>("linear solver maximum iterations"),
+                          this->get_equation_systems().parameters.get<Real>("linear solver tolerance"));
+  else
+    libmesh_error_msg("ERROR: Insufficient linear solver parameters");
```

Comment:
This isn't what we want to do, is it?  If we override one parameter for a particular system, we should still just default to the EquationSystems parameter value for the other.

Reply:
ah that works.
I was thinking of forcing the user to be consistent between where each parameter comes from

it s probably more flexible this other way